### DISCUSSION TITLE
Fetching posts ordered by distance between input coordinates and ACF Google Map field data

### DISCUSSION THREAD

**Discussion by ririshi** [wp-graphql/wp-graphql#2695] [Advanced Custom Fields]
I'm trying to fetch a CPT's posts ordered by distance from a given set of coordinates. The CPT has an ACF Google Map field, which holds a latitude and longitude. I'm trying to order the posts by the haversine distance computed between those coordinates, and the coordinates given as a query argument. I'd like to extend the WPGraphQL schema by either adding a new field on the RootQuery, or by extending the existing post type plural field.

So far, I came up with a way to compute haversine distance as a field I added to the "distributors" by using a custom input type, as well as a `distanceTo` field, added using the `graphql_object_fields` filter. With the query below, I can fetch distributors (the CPT in question), with the distance to the given coordinates, but I cannot order the output by that field without fetching every distributors and sorting the result client-side.

```
query NewQuery($lat: Float, $lng: Float) {
  distributors {
    nodes {
      title
      distanceTo(coordinates: {lat: $lat, lng: $lng})
    }
  }
}
```

I'm hoping someone knows how I can incorporate the distance computation and subsequent sorting in a way where I can get the closest `n` distributors given a set of coordinates.

**Comment by justlevine** [comment]
Hey @Ririshi 

In general, the best forward-compatible way would be to add `DISTANCE` as a possible value on `PostObjectsConnectionOrderbyEnum`, and then use the `graphql_connection_query_args` to set the logic to use with WP_Query natively.

That said, since the haversine distance varies depending on the provided location (which I'm assuming you are also adding as where args on the CPT), you need to change how WPQuery itself works ( [e.g.](https://gist.github.com/akshuvo/4c37df4bd128eb801b7739748ee3cd65). ). If you're creating a custom class like in the example (or using a 3rd party plugin for querying by geoloc), then you'd filter the connection resolver to use your custom class instead of the default WPQuery (see the `graphql_connection_query` filter)
